qqmlsortfilterproxymodel.cpp

```diff
@@ -52,6 +52,24 @@ int QQmlSortFilterProxyModel::count() const
     return rowCount();
 }
 
+/*!
+    \qmlproperty int SortFilterProxyModel::limit
+
+   Limit the total number of results
+*/
+int QQmlSortFilterProxyModel::limit() const
+{
+    return m_limit;
+}
+
+void QQmlSortFilterProxyModel::setLimit(int newLimit)
+{
+    if (m_limit == newLimit)
+        return;
+    m_limit = newLimit;
+    emit limitChanged();
```

Comment:
```suggestion
    emit limitChanged();
    emit countChanged();
```

Reply:
Alternatively, in more elegant in constructor:
```
connect(this, &QAbstractItemModel::limitChanged, this, &QQmlSortFilterProxyModel::countChanged);
```

Reply:
Very good finding! However the count won't change in all cases, so it should be checked and done conditionally.

Reply:
> Very good finding! However the count won't change in all cases, so it should be checked and done conditionally.

Right, something like:
```
void QQmlSortFilterProxyModel::setLimit(int newLimit)
{
    if (m_limit == newLimit)
        return;
        
    const auto currentCount = count();
    m_limit = newLimit;
    emit limitChanged();
    
    if (currentCount != newLimit) {
        emit countChanged();
    }
}
```

Reply:
not ideal, it will be triggered also in situation when count was below limit before and after limit change

Reply:
I think it's safer (and still ok) to `emit countChanged()` together with `limitChanged()`

Reply:
After the fix, the snippet from @osmaczko looks nice. There is no need to emit signals when not needed.

Reply:
OK, will amend that